pkg/provider/kubernetes/ingress/kubernetes.go

```diff
@@ -525,31 +525,34 @@ func loadService(client Client, namespace string, backend networkingv1beta1.Ingr
 		return nil, errors.New("endpoints not found")
 	}
 
-	if len(endpoints.Subsets) == 0 {
+	endpointsLen := len(endpoints.Subsets)
+	if endpointsLen == 0 {
 		return nil, errors.New("subset not found")
 	}
 
 	var port int32
-	for _, subset := range endpoints.Subsets {
+	for i, subset := range endpoints.Subsets {
 		for _, p := range subset.Ports {
 			if portName == p.Name {
 				port = p.Port
 				break
 			}
 		}
 
-		if port == 0 {
+		if port == 0 && i == endpointsLen-1 {
 			return nil, errors.New("cannot define a port")
 		}
 
-		protocol := getProtocol(portSpec, portName, svcConfig)
+		if port != 0 {
```

Comment:
```suggestion
		if port == 0 {
		continue
		}
```

This prevents an additional level indent, and fails fast.